pkg/hooks/hooks_test.go

```diff
@@ -77,3 +79,62 @@ func testManifest() pods.Manifest {
 	builder.SetID(podId)
 	return builder.GetManifest()
 }
+
+func TestHookRunWithTimeout(t *testing.T) {
+	type sleepTest struct {
+		sleep      int
+		timeout    int
+		shouldPass bool
+	}
+
+	var sleepTests = []sleepTest{
+		{1, 2, true},
+		{2, 1, false},
```

Comment:
I had the solution @anthonybishopric proposed in my prototype and it works well enough (I could use nanoseconds as the unit as long as the difference between the two was at least 3 orders of magnitude).  This particular test is complicated by the fact that a Hook has to shell out and uses `sleep`.

`sleep` isn't implemented the same way on every system and sometimes only allows integers so this solves for the least common denominator.  However, `sleep` on macos and RHEL6 *do* support decimal values.  I'm not sure what travis-ci uses, but I can give it a try.